File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       
        int cp= prices[0];
        int profit=0;

        for(int i=1; i <prices.size(); i++){
            profit= max(profit , prices[i]- cp);
            cp= min(cp, prices[i]);
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        n= len(prices)
        cheapest_day=prices[0]
        max_profit= 0

        for i in range(1,n):
            cheapest_day= min(cheapest_day, prices[i])

            max_profit= max(max_profit, prices[i]- cheapest_day)

        return max_profit
            
        

        